pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -87,6 +88,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	err = c.Watch(
 		&source.Kind{Type: &pushv1alpha1.UnifiedPushServer{}},
 		&handler.EnqueueRequestForObject{},
+		predicates.ResourceGenerationOrFinalizerChangedPredicate{},
```

Comment:
@davidffrench I've added this predicate here because since I plagiarised the `manageSuccess` function from the new keycloak operator, it seems like reconciliation was thrashing (each `manageSuccess` call naively updates the status block, and that causes the `metadata.resourceVersion` to change, which causes a new reconciliation, which causes a new reconciliation, ad infinitum)

Reply:
This is correct. The only cluster type I noticed this was an issue on was `minishift`, all other cluster types were clever enough to not update the resourceVersion if there were no changes. 

What are you testing this on?

Reply:
I was testing on minishift. Is it something that was fixed in a particular version of Kubernetes, and that's why you never saw it anywhere else?

Reply:
I was never able to find out what was different on other clusters compared to minishift. Possibly a parameter on the API server. It was not to do  with a specific kubernetes version from what I saw, the same major.minor version on OSD doesn't have this problem.